www/js/lib/abstractBackend.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * abstractBackend.js: Abstract interface to access Wikipedia archives.
+ *
+ * Copyright 2015 Mossroy and contributors
+ * License GPL v3:
+ *
+ * This file is part of Evopedia.
+ *
+ * Evopedia is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Evopedia is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Evopedia (file LICENSE-GPLv3.txt).  If not, see <http://www.gnu.org/licenses/>
+ */
+'use strict';
+define(['title', 'archive', 'zimArchive', 'util', 'jquery'],
+       function(evopediaTitle, evopediaArchive, zimArchive, util, jQuery) {
+
+    function loadArchiveFromDeviceStorage(storage, path) {
+        if (util.endsWith(path, ".zim")) {
+            return new zimArchive.ZIMArchive(storage, path);
+        } else {
+            var archive = new evopediaArchive.LocalArchive();
+            archive.initializeFromDeviceStorage(storage, path);
+            return archive;
+        }
+    };
+    function loadArchiveFromFiles(files) {
+        var archive = new evopediaArchive.LocalArchive();
+        archive.initializeFromArchiveFiles(files);
+        return achive;
+    };
+    /**
+     *  Scans the DeviceStorage for archives
+     *
+     * @param storages List of DeviceStorage instances
+     * @param callbackFunction Function to call with the list of directories where archives are found
+     */
+    function scanForArchives(storages, callbackFunction) {
```

Comment:
I did not manage to make this function work.
It behaves as if it never finds any archive (of any format).
But when I put a breakpoint in it, the archives are in fact found, AFTER the GUI said it did not find anything.
So I suspect an issue about the Promises. Maybe because of the mix between jQuery and Q libraries?